fabric-mining-levels-v0/src/main/java/net/fabricmc/fabric/impl/tools/ToolManager.java

```diff
@@ -87,7 +88,9 @@ public static void registerBreakByTool(Block block, Tag<Item> tag, int miningLev
 	}
 
 	private static int getMiningLevel(ItemStack stack) {
-		if (stack.getItem() instanceof ToolItem) {
+		if (stack.getItem() instanceof DynamicTool) {
+			return ((DynamicTool) stack.getItem()).getMiningLevel(stack);
```

Comment:
Also since the interface is literally just another version of `ToolItem` (below) I'd be inclined to call it `DynamicToolItem`.


Though it's not a class... So maybe `MiningLevelDeterminent` or something, since it doesn't _really_ have to be a tool, either, does it? It's just as valid for a block to implement the interface to be used elsewhere.

Reply:
Perhaps `DynamicToolStats` as suggested in Discord?

Reply:
maybe `DynamicMiningStats` instead, because I'm gonna do a restructure to make this work for any item instead of just tools.